src/navigation-instruction.ts

```diff
@@ -251,9 +266,8 @@ export class NavigationInstruction {
       .all(loads)
       .then(() => {
         delaySwaps.forEach(x => x.viewPort.swap(x.viewPortInstruction));
-        return null;
-      })
-      .then(() => prune(this));
+        prune(this);
```

Comment:
You're skipping a tick here, is that intentional?

Reply:
I think this was an accident converting this fn back and forth between promise and async/await. Thanks!

Reply:
I was confused by this, and did some check, by playing with console. and also trying to read the spec and asking on StackOverflow. From what I've seen so far, chaining seems to be equivalent to synchronous call without any intermediate step.

Reply:
Each `.then()` puts the code that follows in the next tick. So there's definitely a difference between the two.
This would surface when there are other promises going on in parallel (which there probably are). Now prune would come one tick earlier:

```ts
function prune(id) {
	console.log('prune', id);
}
function doSomething(id) {
	console.log('doSomething', id);
}
function pruneNextTick() {
	Promise.resolve()
		.then(() => {
			doSomething(1);
		})
		.then(() => {
			prune(2);
		})
	Promise.resolve()
		.then(() => {
			doSomething(3);
		})
}
function pruneSameTick() {
	Promise.resolve()
		.then(() => {
			doSomething(1);
			prune(2);
		})
	Promise.resolve()
		.then(() => {
			doSomething(3);
		})
}
```

`pruneNextTick() // simulates previous code`

```shell
doSomething 1
doSomething 3
prune 2
```

`pruneSameTick() // simulates current code`

```shell
doSomething 1
prune 2
doSomething 3
```

Whether it matters / breaks something is another question. Perhaps it doesn't matter. But there's a difference for sure

Reply:
```js
Promise.resolve()
		.then(() => {
			doSomething(1);
		})
		.then(() => {
			prune(2);
		})
	Promise.resolve()
		.then(() => {
			doSomething(3);
		})
```
Isn't the the log the way it is because the way promise callback are enqueued, not because of the tick behavior? I'd expect at it to log exactly as your example showed.

Reply:
You can write results to an array, result is the same. Regardless, whoever told you that this:
```ts
.then(() => a())
.then(() => b())
```
it the same as this:
```ts
.then(() => a(); b();)
```

Is wrong. If you asked on SO and they gave you this answer, tell me the link to the question and I'll tell them how wrong they are.

Reply:
I just realized I slightly misread your message. I thought you implied there was different behavior when logging in the console.
I'm not sure what it matters whether it's about the way promises are queued vs the tick behavior. Aren't those 2 things the same fabric anyway?
Each `.then` is placed on the micro task queue after the previous code is done executing. If that previous code is synchronous, then it happens the next tick. Even if you put 1000 `Promise.resolve().then()` calls right below eachother (not chained), all 1000 will happen the next tick. But if you chain only one, the second `then()` happens the tick after the previous `then()`, so 2 ticks later..